docs/rationale.md

```diff
@@ -0,0 +1,180 @@
+above are also applied within this directory.
+
+[The `tests/harness`
+directory](https://github.com/tc39/test262/tree/main/test/harness) stores tests
+for the "harness" files which Test262 maintains to assist in test writing.
+
+Directories are not generally applied beyond these limits; further
+differentiation is instead achieved through structured file names which follow
+ad-hoc conventions. This organization balances the need to group tests
+logically with the need to discover tests. See, for example, [the tests for
+template
+literals](https://github.com/tc39/test262/tree/main/test/language/expressions/template-literal).
+
+Many consumers use file names as a way to compare test results across revisions
+and between implementations. For this reason, tests files are rarely
+re-organized after being accepted.
+
+## Regression tests
+
+It is possible to write tests for semantics which, while not explicitly
+specified by ECMA262, are nonetheless valid according to the normative text.
+Such tests are welcome in Test262, but their fitness is not a given. Consumers
+from many constituencies value the coherence and consistency of the test suite,
+and tests which disallow arbitrary extraneous behavior can degrade those
+qualities. Because Test262 is not maintained as a repository of regression
+tests, contributions which include these kinds of tests will be weighed against
+their likelihood of identifying error in a plurality of implementations.
+
+For example, assume that some runtime spuriously accesses the `toJSON` property
+of the value passed to
+[`String.prototype.repeat`](https://tc39.es/ecma262/#sec-string.prototype.repeat).
+While the maintainers of the engine may decide to include a regression test
+which disallows that behavior in their project, the maintainers of Test262
+would not necessarily accept such a test.
+
+## Large tests
+
+Test262 tests are typically very focused. The vast majority exercise just one
+algorithm step/grammar production, and some are even more granular than that!
+
+Some test contributors are uncomfortable splitting their work across files like
+this. It's certainly unlike the practices that are common in modern application
+development. In those settings, many tests are often grouped into the same file
+and separated by function boundaries.
+
+Test262 doesn't use the same approach as a typical application test suite in
+order to limit complexity. The guidance of "one test per file" means that
+consuming Test262 is relatively easy; there is no "test runner" API for
+consumers to implement, and interpreting results is likewise straightforward.
+It also lowers the barrier to entry for new contributors since there is no API
+to learn.
+
+## Syntax tests
+
+When testing a syntactic feature of the language, it can be tempting to write
+tests which verify that some bit of source text does *not* produce a syntax
+error. Contributors should try to push beyond verifying only the lack of a
```

Comment:
It _can_ be tempting to write such tests, and this document should explain and attempt to justify the proper response to that temptation.

Reply:
I see what you're saying, and I think the first statement is just missing an operative word. Of course we want tests for syntactically-interesting forms, but the statement "it can be tempting to [verify that runtimes successfully parse some syntax]" heavily insinuates there's something wrong with the practice. I think a qualifier like "only" is critical:

> When testing a syntactic feature of the language, it can be tempting to write tests which verify only that some bit of source text does *not* produce a syntax error.

Thanks for catching this!